CMakeLists.txt

```diff
@@ -18,8 +18,15 @@ check_struct_has_member("struct msghdr" "msg_flags" "sys/types.h;sys/socket.h" H
 set(CMAKE_EXTRA_INCLUDE_FILES "sys/types.h" "sys/socket.h")
 check_type_size("socklen_t" HAS_SOCKLEN_T BUILTIN_TYPES_ONLY)
 unset(CMAKE_EXTRA_INCLUDE_FILES)
+
+set(ENET_BUILD_LIBTYPE "STATIC" CACHE STRING "ENet library build type. (SHARED/STATIC)")
```

Comment:
Please don't use a custom option but the standard `BUILD_SHARED_LIBS`

https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS